include/bmi/Bmi_Adapter.hpp

```diff
@@ -20,29 +20,13 @@ namespace models {
         public:
 
             Bmi_Adapter(std::string model_name, std::string bmi_init_config, std::string forcing_file_path, bool allow_exceed_end,
-                        bool has_fixed_time_step, utils::StreamHandler output)
-                : model_name(std::move(model_name)),
-                  bmi_init_config(std::move(bmi_init_config)),
-                  bmi_model_uses_forcing_file(!forcing_file_path.empty()),
-                  forcing_file_path(std::move(forcing_file_path)),
-                  bmi_model_has_fixed_time_step(has_fixed_time_step),
-                  allow_model_exceed_end_time(allow_exceed_end),
-                  output(std::move(output)),
-                  bmi_model_time_convert_factor(1.0)
-            {
-                // This replicates a lot of Initialize, but it's necessary to be able to do it separately to support
-                // "initializing" on construction, given using Initialize requires use of virtual functions
-                errno = 0;
-                if (!utils::FileChecker::file_is_readable(this->bmi_init_config)) {
-                    init_exception_msg = "Cannot create and initialize " + this->model_name + " using unreadable file '"
-                            + this->bmi_init_config + "'. Error: "+std::strerror(errno);
-                    throw std::runtime_error(init_exception_msg);
-                }
-            }
+                        bool has_fixed_time_step, utils::StreamHandler output);
 
             Bmi_Adapter(Bmi_Adapter const&) = delete;
             Bmi_Adapter(Bmi_Adapter &&) = default;
 
+            virtual ~Bmi_Adapter() = default;
```

Comment:
For clarity, this is set to default instead of calling `Finalize()` to prevent undefined behavior, since `Finalize()` is virtual.

Reply:
Should this then be a pure virtual function so that subclasses are required to implement a destructor (even if it is the default)?  We would have to double check the exact usage of `Bmi_Adapter` and make sure we aren't using it as a concrete type anywhere, but might be worth thinking about.

Reply:
It can't be pure virtual because `Bmi_Adapter` has member variables, so you'll get a linker error on compilation, i.e. https://godbolt.org/z/o9rK41qdT

Reply:
After a brief discussion, this is actually more of a syntactical choice. We can define `~Bmi_Adapter() = 0`, and then implement `= default` in its source file to make it explicit that derived classes *should* implement a destructor, but there's no way to force that.

Reply:
Added in [e2678b6](https://github.com/NOAA-OWP/ngen/pull/750/commits/e2678b61e43188ce517202a43c0ce3a06321740f)